models/server-attachment/src/index.ts

```diff
@@ -24,6 +25,11 @@ export { serverAttachmentId } from '@hcengineering/server-attachment'
 
 export function createModel (builder: Builder): void {
   builder.createDoc(serverCore.class.Trigger, core.space.Model, {
-    trigger: serverAttachment.trigger.OnAttachmentDelete
+    trigger: serverAttachment.trigger.OnAttachmentDelete,
+    txMatch: {
+      _class: core.class.TxCollectionCUD,
+      'tx.objectClass': attachment.class.Attachment,
```

Comment:
We also have attachment.class.Photo and document.class.CollaboratorDocument which derived from Attachment. We need think about good solution for it. @haiodo do you have any idea? Probably we should have another field to check object class.

Reply:
Best ideas will be to make platform for every Ref<Class<Doc>> field treat them as class and all derived from it in case of matching. We do it manually for findAll for _class field, so it we do so for every reference to Class, it should simplify places like this. But we will need type for tx, and we do not have it right now, so right now in triggers we need to update query and if we have filter for tx.objectClass, we should extend it to any derived from class properly.